src/Features/VisualBasic/Portable/IntroduceVariable/VisualBasicIntroduceVariableService_IntroduceLocal.vb

```diff
@@ -112,49 +112,37 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.IntroduceVariable
             Return Nothing
         End Function
 
-        Private Async Function ComplexifyNextStatementAsync(
-                document As Document,
-                localAnnotation As SyntaxAnnotation,
-                cancellationToken As CancellationToken) As Task(Of Document)
-
-            ' Retrieve the declarationStatement we just inserted and complexify the next statement.
-            Dim oldRoot = Await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(False)
-            Dim declarationStatement = oldRoot.GetAnnotatedNodes(Of LocalDeclarationStatementSyntax)(localAnnotation).Single()
-
-            Dim nextStatement = declarationStatement.GetNextStatement()
-            If nextStatement IsNot Nothing Then
-                Dim complexifiedNextStatement = Await Simplifier.ExpandAsync(nextStatement, document, cancellationToken:=cancellationToken).ConfigureAwait(False)
-                complexifiedNextStatement = complexifiedNextStatement.WithAdditionalAnnotations(Formatter.Annotation)
-
-                Dim newRoot = oldRoot.ReplaceNode(nextStatement, complexifiedNextStatement)
-                Return document.WithSyntaxRoot(newRoot)
-            End If
-
-            Return document
-        End Function
-
         Private Async Function IntroduceLocalDeclarationIntoBlockAsync(
                 document As SemanticDocument,
                 container As SyntaxNode,
                 expression As ExpressionSyntax,
                 newLocalName As NameSyntax,
-                localDeclaration As LocalDeclarationStatementSyntax,
+                declarationStatement As LocalDeclarationStatementSyntax,
                 allOccurrences As Boolean,
                 cancellationToken As CancellationToken) As Task(Of Document)
 
             Dim localAnnotation = New SyntaxAnnotation()
-            localDeclaration = localDeclaration.WithAdditionalAnnotations(Formatter.Annotation, localAnnotation)
+            declarationStatement = declarationStatement.WithAdditionalAnnotations(Formatter.Annotation, localAnnotation)
 
             Dim oldOutermostBlock = container
             If oldOutermostBlock.IsSingleLineExecutableBlock() Then
                 oldOutermostBlock = oldOutermostBlock.Parent
             End If
 
             Dim matches = FindMatches(document, expression, document, oldOutermostBlock, allOccurrences, cancellationToken)
+
+            Dim complexified = Await ComplexifyParentingStatements(document, matches, cancellationToken).ConfigureAwait(False)
```

Comment:
Before the change, the complexification was happening after the local was introduced. But that is too late, as we need the original syntax to make names explicit in tuples and anonymous types.
The new flow mirrors the C# implementation of IntroduceVariable.